src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentState.ComputedStateTracker.cs

```diff
@@ -0,0 +1,253 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+internal partial class DocumentState
+{
+    // Internal, because we are temporarily sharing code with CohostDocumentSnapshot
+    private class ComputedStateTracker
```

Comment:
All of this green made me remember how much I hate how confusing `ComputedStateTracker` is and how much I want to rewrite it. Sigh... MUST RESIST BEING NERD-SNIPED. 😁